modules/WordCloudPlugin/src/main/java/net/clementlevallois/wordcloud/TopTermExtractor.java

```diff
@@ -0,0 +1,186 @@
+import net.clementlevallois.umigon.model.TextFragment;
+import net.clementlevallois.umigon.model.TypeOfTextFragment.TypeOfTextFragmentEnum;
+import net.clementlevallois.umigon.tokenizer.controller.UmigonTokenizer;
+import net.clementlevallois.utils.Multiset;
+import org.gephi.graph.api.Column;
+import org.gephi.graph.api.Graph;
+import org.gephi.graph.api.GraphModel;
+import org.gephi.graph.api.Node;
+import org.gephi.graph.api.NodeIterable;
+import org.gephi.io.importer.api.Container;
+import org.gephi.io.importer.api.ContainerUnloader;
+import org.gephi.io.importer.api.ImportController;
+import org.gephi.io.importer.plugin.file.ImporterGEXF;
+import org.gephi.io.importer.spi.FileImporter;
+import org.gephi.io.processor.plugin.DefaultProcessor;
+import org.gephi.project.api.ProjectController;
+import org.openide.util.Exceptions;
+import org.openide.util.Lookup;
+
+/**
+ *
+ * @author LEVALLOIS
+ */
+public class TopTermExtractor {
+
+    public static void main(String args[]) throws IOException, Exception {
+        new TopTermExtractor().importFromFile();
+    }
+
+    public void importFromFile() throws IOException, Exception {
+
+        Path exampleGexf = Path.of("G:\\Mon Drive\\Twitch stream\\gephi plugin development\\gephi-plugins\\modules\\wordcloudPlugin\\qatar user network.gexf");
+
+        String gexfFileAsString = Files.readString(exampleGexf, StandardCharsets.UTF_8);
+
+        //Init a project - and therefore a workspace
+        ProjectController pc = Lookup.getDefault().lookup(ProjectController.class);
+        pc.newProject();
+
+        //Get controllers and models
+        ImportController importController = Lookup.getDefault().lookup(ImportController.class);
+
+        //Import file
+        Container container;
+        FileImporter fi = new ImporterGEXF();
+
+        //Append imported data to GraphAPI
+        container = importController.importFile(new StringReader(gexfFileAsString), fi);
+        container.closeLoader();
+
+        DefaultProcessor processor = new DefaultProcessor();
+        processor.setWorkspace(pc.getCurrentWorkspace());
+        processor.setContainers(new ContainerUnloader[]{container.getUnloader()});
+        processor.process();
+
+    }
+
+    public Boolean tokenizeSelectedTextualAttributeForTheEntireGraph(GraphModel gm, String attributeName, String lang) {
+
+        Graph graph = gm.getGraph();
+
+        // selecting the column corresponding to the attribute we want to analyze
+        Column attributeToBeAnalyzed = gm.getNodeTable().getColumn(attributeName);
+
+        // we will iterate / loop on the nodes of the graph
+        NodeIterable nodes = graph.getNodes();
+
+        // we will store the text of the attribute for each node in a list
+        Map<String, String> textsFromTheAttribute = new HashMap();
+
+
+        // doing the iteration now
+        Iterator<Node> iteratorOnNodes = nodes.iterator();
+        while (iteratorOnNodes.hasNext()) {
```

Comment:
Using a for loop is more standard here
```
for(Node node : nodes) {
}
```

Reply:
I have a fondness for Iterators but indeed it is an overkill there